File: src/cpp/Source/Chunks/Common/ProgramType.cpp
#include "PCH.h"
#include "ProgramType.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(ProgramType value)
{
	switch (value)
	{
	case ProgramType::PixelShader :
		return "ps";
	case ProgramType::VertexShader :
		return "vs";
	case ProgramType::GeometryShader :
		return "gs";
	case ProgramType::HullShader :
		return "hs";
	case ProgramType::DomainShader :
		return "ds";
	case ProgramType::ComputeShader :
		return "cs";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}

File: src/SlimShader/Chunks/Common/ProgramType.cs
namespace SlimShader.Chunks.Common
{
	public enum ProgramType : ushort
	{
		[Description("ps")]
		PixelShader = 0,

		[Description("vs")]
		VertexShader = 1,

		[Description("gs")]
		GeometryShader = 2,

		// Below are shaders new in DX 11

		[Description("hs")]
		HullShader = 3,

		[Description("ds")]
		DomainShader = 4,

		[Description("cs")]
		ComputeShader = 5
	}
}